src/main/java/ca/gc/cyber/ops/assemblyline/java/client/clients/AssemblylineClient.java

```diff
@@ -129,16 +146,41 @@ public AssemblylineClient withAuthBearerToken(String authBearerToken) {
         return newClient;
     }
 
-    protected void buildWebClient(AssemblylineClientProperties assemblylineClientProperties) {
+    protected void buildWebClient(AssemblylineClientProperties assemblylineClientProperties,
+                                  ProxyProperties proxyProperties) {
 
-        webClient = WebClient.builder()
+        HttpClient httpClient = createHttpClient(proxyProperties);
+
+        webClient = WebClient.builder().clientConnector(new ReactorClientHttpConnector(httpClient))
                 .codecs(clientCodecConfigurer -> {
-                    clientCodecConfigurer.defaultCodecs().jackson2JsonDecoder(new Jackson2JsonDecoder(this.mapper));
-                    clientCodecConfigurer.defaultCodecs().jackson2JsonEncoder(new Jackson2JsonEncoder(this.mapper));
-                })
-                .filter(addSession)
-                .baseUrl(assemblylineClientProperties.getUrl())
-                .build();
+                    clientCodecConfigurer.defaultCodecs().jackson2JsonDecoder(new Jackson2JsonDecoder(mapper));
+                    clientCodecConfigurer.defaultCodecs().jackson2JsonEncoder(new Jackson2JsonEncoder(mapper));
+                }).filter(addSession).baseUrl(assemblylineClientProperties.getUrl()).build();
+    }
+
+    /**
+     * Return an HttpClient that uses the proxy setup via the properties, if properties host and port are both set. If
+     * no proxy properties are set, a 'secure' Httpclient is returned without proxy configured.
+     *
+     * @return HttpClient
+     * @throws IllegalArgumentException If proxy port is not valid
+     */
+    private static HttpClient createHttpClient(ProxyProperties proxyProperties) {
+        String proxyHost = proxyProperties.getHost();
+        int proxyPort = proxyProperties.getPort();
+
+        if (proxyHost == null) {
+            log.debug("No proxy host set. Assembly line client not configured to go through a proxy.");
+            return HttpClient.create().secure();
+        }
+
+        if (proxyPort <= 0) {
```

Comment:
Instead of validating here you can use spring validation (@Validated) with @Min(1) directly on the property class, https://docs.spring.io/spring-boot/docs/2.1.13.RELEASE/reference/html/boot-features-external-config.html#boot-features-external-config-validation